src/com/company/GameManager.java

```diff
@@ -36,58 +46,44 @@ public static void play() {
         for (int i = 0; i < NUMBER_OF_PLAYERS; i++) {
             ioManager.showMessage(format(FORMATTED_PROVIDE_NAME, PLAYER, i + 1, PROVIDE_YOUR_NAME));
             String name = ioManager.getName();
-
             players[i] = new Player(name, sign[i]);
-
             ioManager.showMessage(format(FORMATED_WELCOME, HELLO, players[i].getName(), PLAYING_AS, players[i].getSign()));
-
         }
+    }
 
+    public void playGame(Player players[]) {
+        boolean shouldPlayAgain = true;
         final int boardSize = ioManager.getBoardSize(MIN_BOARD_SIZE, MAX_BOARD_SIZE);
         ioManager.showMessage(format(FORMATED_SELECT, SELECTED, boardSize));
         final Board board = new Board(boardSize);
-
-        Player currentPlayer = players[0];
-        if (getRandomPlayerIndex(NUMBER_OF_PLAYERS) == 1) {
-            currentPlayer = players[1];
-        }
+        int activePlayerId = getRandomPlayerIndex(NUMBER_OF_PLAYERS);
 
         ioManager.showMessage(DRAWN_PLAYER);
-        ioManager.showMessage(format(FORMATED_PLAYER, currentPlayer.getName(), currentPlayer.getSign()));
+        ioManager.showMessage(format(FORMATED_PLAYER, players[activePlayerId].getName(), players[activePlayerId].getSign()));
 
         do {
             ioManager.showBoard(board);
-            ioManager.showMessage(format(FORMATED_PLAYER, currentPlayer.getName(), currentPlayer.getSign()));
+            ioManager.showMessage(format(FORMATED_PLAYER, players[activePlayerId].getName(), players[activePlayerId].getSign()));
 
             final Coordinates coordinates = ioManager.getCoordinates(MIN_BOARD_ID, boardSize);
 
-            System.out.printf(FORMATED_CHOICE, SELECTED, ROW, coordinates.row + 1, COLUMN, coordinates.column + 1);
+            ioManager.showMessage(format(FORMATED_CHOICE, SELECTED, ROW, coordinates.row + 1, COLUMN, coordinates.column + 1));
 
-            if (board.insertSign(currentPlayer.getSign(), coordinates.row, coordinates.column)) {
+            if (board.insertSign(players[activePlayerId].getSign(), coordinates.row, coordinates.column)) {
```

Comment:
dla skrócenia zapisu `players[activePlayerId]` możesz w pierwszej linii pętli stworzyć zmienną i przypisać aktywnego gracza: `final Player player = players[activePlayerId]` nie tworzy się tutaj nowy obiekt więc taki zapis w pętli jest spoko.
Doda to jedną nadmiarową linijkę, ale skróci zapis co najmniej trzech innych.
Oczywiście to już kosmetyka więc zrób jak uważasz.

Reply:
zrobiłam (a właściwie wróciłam bo całkiem podobnie już było) bo zapis jest rzeczywiście krótszy,